docs/dev/pagination-v2-follow-up_a.md

```diff
@@ -0,0 +1,81 @@
+# Pagination in v2 SQL Engine
+
+Implementing pagination in the OpenSearch SQL plugin can be partitioned along several dimensions:
+1. SQL query type,
+1. Size of queried dataset,
+1. Number of open cursors,
+1. SQL node load balancing and failover.
+
+## SQL Query Type
+The two main query types are filter-and-project and aggregation queries. 
+
+V1 engine only supports cursors for filter-and-project queries.
+
+Therefore, discussion of pagination in V2 engine will also be limited to filter-and-project queries.
+
+Pagination of aggregation queries can be considered after V1 engine is deprecated.
+
+## Size of Queried Dataset
+
+OpenSearch provides several data retrival APIs that are optimized for different use cases.
+
+At this time, SQL plugin concerns only uses simple search API and scroll API.
+
+Simple search API will only return at most `window_size` number of documents where `window_size` is an index setting.
```

Comment:
index setting?  
You mean an OpenSearch configuration setting?

Reply:
That's OpenSearch terminology for a setting that can be configured for each index.

I got the name wrong though, should've been `max_result_window`.